extensions/integration-tests/src/test/schemaCompare.test.ts

```diff
@@ -270,6 +274,231 @@ suite('Schema compare integration test suite @DacFx@', () => {
+
+		assert(schemaCompareResult.success === false, `Expected: failure in schema compare. Actual: Success`);
+		assert(schemaCompareCancelResult.success === true, `Expected: success in schema compare cancel. Actual: Failure`);
+	});
+
+	test('Schema compare dacpac to database comparison with publishing some changes and then compare again', async function () {
+		this.timeout(5 * 60 * 1000);
+		let server = await getStandaloneServer();
+		const ownerUri = await getConnectionUri(server);
+		const now = new Date();
+		const operationId = 'testOperationId_' + now.getTime().toString();
+		const targetDB: string = 'ads_schemaCompare_targetDB_' + now.getTime().toString();
+
+		try {
+			assert(dacfxService, 'DacFx Service Provider is not available');
+			let result = await dacfxService.deployDacpac(dacpac1, targetDB, true, ownerUri, azdata.TaskExecutionMode.execute);
+
+			assert(result.success === true, 'Deploy database 2 (target) should succeed');
+
+			const source: mssql.SchemaCompareEndpointInfo = {
+				endpointType: mssql.SchemaCompareEndpointType.Dacpac,
+				packageFilePath: dacpac2,
+				serverDisplayName: '',
+				serverName: '',
+				databaseName: '',
+				ownerUri: ownerUri,
+				connectionDetails: undefined
+			};
+			const target: mssql.SchemaCompareEndpointInfo = {
+				endpointType: mssql.SchemaCompareEndpointType.Database,
+				packageFilePath: '',
+				serverDisplayName: '',
+				serverName: server.serverName,
+				databaseName: targetDB,
+				ownerUri: ownerUri,
+				connectionDetails: undefined
+			};
+
+			assert(schemaCompareService, 'Schema Compare Service Provider is not available');
+
+			const schemaCompareResult = await schemaCompareService.schemaCompare(operationId, source, target, azdata.TaskExecutionMode.execute, null);
+			assertSchemaCompareResult(schemaCompareResult, operationId, 4);
+
+			//try to exclude the difference of function Function1 and it should not fail
+			let function1Difference = schemaCompareResult.differences.find(e => e.sourceValue && e.sourceValue[1] === 'Function1' && e.name === 'SqlInlineTableValuedFunction');
+			assert(function1Difference !== undefined, 'The difference function Function1 should be found. Should not be undefined');
+
+			//by default all changes are included while publishing, hence excluding Function1
+			const includeResult = await schemaCompareService.schemaCompareIncludeExcludeNode(operationId, function1Difference, false, azdata.TaskExecutionMode.execute);
+			assertIncludeExcludeResult(includeResult, true, 0, 0);
+
+			//publish the updated changes. Function1 should not be added to the target database
+			const publishChangesResult = await schemaCompareService.schemaComparePublishChanges(schemaCompareResult.operationId, server.serverName, targetDB, azdata.TaskExecutionMode.execute);
+			assert(publishChangesResult.success === true, 'Publish changes should complete successfully. But it failed.');
+
+			//verify table Table3 is added
+			const retryCount = 24; // 2 minutes
```

Comment:
does it need to be this long? Shouldn't the table be created after the schemaComparePublishChanges() finishes?

Reply:
Also you already define this above - do you really need to define it again down here?

Reply:
It does need a few retry cycles, but may be not for this long. Reduced it to 1min.

And removed the extra declaration here.